### PR TITLE
[zephyr-exporter] Property for export executions with specific statuses

### PR DESCRIPTION
**Pull request by abudevich**

### CODE REVIEW

File: vividus-to-zephyr-exporter/src/main/java/org/vividus/zephyr/parser/TestCaseParser.java [hunk 1]
```diff
@@ -62,10 +62,26 @@ public TestCaseParser(ZephyrExporterProperties zephyrExporterProperties)
...
         }
+        testCasesForImporting = filterByStatuses(testCasesForImporting);
         LOGGER.info("Test cases for exporting to JIRA: {}", testCasesForImporting);
         return testCasesForImporting;
     }
 
+    private List<TestCase> filterByStatuses(List<TestCase> testCases)
+    {
+        List<TestCaseStatus> statusesForExport = zephyrExporterProperties.getStatusesForExport();
+        if (statusesForExport.isEmpty())
```
Comment (uarlouski) [inline vividus-to-zephyr-exporter/src/main/java/org/vividus/zephyr/parser/TestCaseParser.java:73]: statusesForExport sounds like we should export only test cases with such statuses, but if this is empty we export all test cases, maybe set some default value?

File: vividus-to-zephyr-exporter/src/main/java/org/vividus/zephyr/parser/TestCaseParser.java [hunk 2]
```diff
@@ -62,10 +62,26 @@ public TestCaseParser(ZephyrExporterProperties zephyrExporterProperties)
...
+        {
+            return testCases;
+        }
+        else
```
Comment (uarlouski) [inline vividus-to-zephyr-exporter/src/main/java/org/vividus/zephyr/parser/TestCaseParser.java:77]: redundant else

File: vividus-to-zephyr-exporter/src/main/java/org/vividus/zephyr/parser/TestCaseParser.java (hunk 1 above)
Comment (abudevich) [reply vividus-to-zephyr-exporter/src/main/java/org/vividus/zephyr/parser/TestCaseParser.java:73]: Anything like `zephyr.exporter.statuses-for-export=FAILD,BROKEN,PENDING,UNKNOWN,SKIPPED,PASSED` ?

File: vividus-to-zephyr-exporter/src/main/resources/application.properties [hunk 3]
```diff
@@ -35,3 +35,5 @@ zephyr.statuses.PENDING=TEST NEEDS REWORK
 zephyr.statuses.UNKNOWN=PASSED WITH REMARKS
 zephyr.statuses.SKIPPED=SCHEDULED
 zephyr.statuses.PASSED=PASSED
+
+zephyr.exporter.statuses-for-export=
```
Comment (valfirst) [inline vividus-to-zephyr-exporter/src/main/resources/application.properties:39]:
```
zephyr.exporter.statuses-of-test-cases-to-add-to-execution
```

File: vividus-to-zephyr-exporter/src/main/java/org/vividus/zephyr/parser/TestCaseParser.java [hunk 4]
```diff
@@ -62,10 +62,26 @@ public TestCaseParser(ZephyrExporterProperties zephyrExporterProperties)
         {
             testCasesForImporting.add(new TestCase(entry.getKey(), entry.getValue().first()));
         }
+        testCasesForImporting = filterByStatuses(testCasesForImporting);
```
Comment (valfirst) [inline vividus-to-zephyr-exporter/src/main/java/org/vividus/zephyr/parser/TestCaseParser.java:65]: `Importing` -> `Exporting`

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-to-zephyr-exporter/src/main/java/org/vividus/zephyr/parser/TestCaseParser.java [hunk 5]
```diff
@@ -57,13 +57,21 @@ public TestCaseParser(ZephyrExporterProperties zephyrExporterProperties)
                 .collect(Collectors.groupingBy(TestCase::getKey,
                         Collectors.mapping(TestCase::getStatus, Collectors.toCollection(TreeSet::new))));
 
-        List<TestCase> testCasesForImporting = new ArrayList<>();
+        List<TestCase> testCasesForExporting = new ArrayList<>();
         for (Map.Entry<String, TreeSet<TestCaseStatus>> entry : testCasesMap.entrySet())
         {
-            testCasesForImporting.add(new TestCase(entry.getKey(), entry.getValue().first()));
+            testCasesForExporting.add(new TestCase(entry.getKey(), entry.getValue().first()));
```
Comment (valfirst) [inline vividus-to-zephyr-exporter/src/main/java/org/vividus/zephyr/parser/TestCaseParser.java:63]:
```
TestCaseStatus testCaseStatus = entry.getValue().first();
if (zephyrExporterProperties.getStatusesOfTestCasesToAddToExecution().contains(testCaseStatus))
{
    testCasesForExporting.add(new TestCase(entry.getKey(), testCaseStatus));
}

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

**Review by valfirst — APPROVED** [review]

File: docs/modules/integrations/pages/zephyr-exporter.adoc [hunk 6]
```diff
@@ -24,6 +24,10 @@ Features:
 |false
 |Property for update existing executions statuses only.
 
+|`zephyr.exporter.statuses-of-test-cases-to-add-to-execution`
+|true
```
Comment (uarlouski) [inline docs/modules/integrations/pages/zephyr-exporter.adoc:28]: why is it required if it has default val?

File: vividus-to-zephyr-exporter/src/test/java/org/vividus/zephyr/parser/TestCaseParserTests.java [hunk 7]
```diff
@@ -111,19 +115,39 @@ void testResultsProblemWithReading() throws URISyntaxException
...
+    void testCreateTestCasesWithStatusFilter() throws URISyntaxException, IOException
+    {
+        configureObjectMapper();
+        Path sourceDirectory = Paths.get(getClass().getResource(RESOURCE_PATH).toURI());
+        when(zephyrExporterProperties.getSourceDirectory()).thenReturn(sourceDirectory);
+        when(zephyrExporterProperties.getStatusesOfTestCasesToAddToExecution())
+            .thenReturn(List.of(TestCaseStatus.PASSED));
+        List<TestCase> testCases = testCaseParser.createTestCases(objectMapper);
+        assertEquals(testCases.size(), 1);
+        assertThat(testLogger.getLoggingEvents().get(0).getMessage(), is(JSON_FILES_STRING));
```
Comment (uarlouski) [inline vividus-to-zephyr-exporter/src/test/java/org/vividus/zephyr/parser/TestCaseParserTests.java:143]: testLogger.getLoggingEvents() cache

**Review by uarlouski — APPROVED** [review]

**Pull request merged**